Sources/SwiftDocC/Infrastructure/Diagnostics/ANSIAnnotation.swift

```diff
@@ -0,0 +1,56 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2023 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+struct ANSIAnnotation {
+    enum Color: UInt8 {
+        case red = 31
+        case green = 32
+        case yellow = 33
+        case `default` = 39
+    }
+    
+    enum Trait: UInt8 {
+        case normal = 0
+        case bold = 1
+        case italic = 3
+    }
+    
+    private var color: Color
+    private var trait: Trait
+    
+    /// The textual representation of the annotation.
+    private var code: String {
+        "\u{001B}[\(trait.rawValue);\(color.rawValue)m"
+    }
+    
+    init(color: Color, trait: Trait = .normal) {
+        self.color = color
+        self.trait = trait
+    }
+    
+    func applied(to message: String) -> String {
+        "\(code)\(message)\(ANSIAnnotation.normal.code)"
+    }
+    
+    static var normal: ANSIAnnotation {
+        self.init(color: .default, trait: .normal)
```

Comment:
nit: this is used to reset the formatting but "0" already resets the color so adding "39" is redundant.

Reply:
Fixed [here](https://github.com/apple/swift-docc/pull/535/commits/25468c701eeac7803bb03fd731af927299ceaee8) thank you!